caddytest/integration/caddyfile_test.go

```diff
@@ -484,3 +485,20 @@ func TestValidPrefix(t *testing.T) {
 		caddytest.AssertAdapt(t, successCase.rawConfig, "caddyfile", successCase.expectedResponse)
 	}
 }
+
+func TestUriReplace(t *testing.T) {
+	tester := caddytest.NewTester(t)
+
+	tester.InitServer(`
+	{
+		admin localhost:2999
+		http_port     3090
+	}
+	:3090
+	uri replace "\}" %7D
+	uri replace "\{" %7B
+	
+	respond "{query}"`, "caddyfile")
+
+	tester.AssertPostResponseBody("http://localhost:3090/endpoint?test={%20content%20}", []string{}, bytes.NewBuffer([]byte{}), 200, "test=%7B%20A%20%7D")
```

Comment:
The use of `AssertPostResponseBody` complicates the test line. A simple GET request is sufficient because we  only want to inspect the response body.